### PR TITLE
initial creation of libcugraph_etl.so

### PR DESCRIPTION
**Pull request by ChuckHastings**
The new renumbering implementation will require C++ integration directly with cudf.  In order to facilitate that, but also support our customers that won't need cudf, this PR will create a separate library (`libcugraph_etl.so`) which will ultimately link with `libcudf.so` and contain the ETL portions of cugraph that require cudf features.

This way our other libcugraph customers that don't need to reference the new library will not need to install all of the cudf dependencies.

To seed this, the PR also includes a proposed API for the new renumbering capability.

### CODE REVIEW

**Review by seunghwak — APPROVED** [review]

**Comment by ChuckHastings** [comment]
rerun tests

File: cpp/include/cugraph_etl/functions.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *                    or INT64.
+ *
+ * @return tuple with the following three values:
+ *    1) column (of type dtype) with the source vertices represented as integers
+ *    2) column (of type dtype) with the destination vertices represented as integers
+ *    3) table with the vertex id as a column and the vertex columns from src_table
+ *       and/or dst_table that correspond to the vertex id
+ *
+ */
+std::tuple<cudf::column, cudf::column, cudf::table> renumber_cudf_tables(
```
[the remark this answers is not in the document]
Comment (ChuckHastings) [reply cpp/include/cugraph_etl/functions.hpp]: Done

**Comment by ChuckHastings** [comment]
I think I have addressed the suggestions.  I'd like to merge this soon if there are no other concerns.

**Review by chirayuG-nvidia — APPROVED** [review]
LGTM

**Review by rlratzel — APPROVED** [review]

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by rlratzel** [comment]
rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by rlratzel** [comment]
rerun tests

Want to make sure the latest cupy is being used.

**Comment by Ethyling** [comment]
Hey, I think you also need to update the [upload script](https://github.com/rapidsai/cugraph/blob/branch-21.12/ci/cpu/upload.sh) to upload the new package

**Review by Ethyling — APPROVED** [review]

**Comment by rlratzel** [comment]
@gpucibot merge

**Pull request merged**